dotcom-rendering/src/model/enhance-placeholders.ts

```diff
@@ -0,0 +1,84 @@
+import { Design } from '@guardian/types';
+import { decideDesign } from '@root/src/web/lib/decideDesign';
+
+export const replacePlaceholders = (
+	elem: InteractiveAtomBlockElement,
+	variables: Map<string, string>,
+): InteractiveAtomBlockElement => {
+	if (!elem.html) {
+		return elem;
+	}
+
+	const replacer = (substring: string, name: string): string => {
+		return variables.get(name) || substring;
```

Comment:
In the case where someone enters something unsupported, e.g "{{ test }}", this is then replaced with "test". Is this preferable over leaving it as "{{ test }}"? 

My concern being that there may be other places where this `{{ }}` syntax is used, and we're not being friendly to other libraries / code by overwriting all instances of it in an article. Especially since we're targeting the whole doc, so this could be in a HTML data parameter, an inline script, etc. 

I don't think it has a major impact on the end user, but leaving it as {{ test }} might also help someone creating an interactive notice that they made a typo, etc.

Reply:
Resolving as I misunderstood the code.

Reply:
@OllysCoding in fact, if we don't have the variable defined as supported (in the passed in variables map) then we don't modify the string at all. One of the tests aims to capture this.